### PR TITLE
PIPRES-180: Order status soft delete on uninstall

### PR DESCRIPTION
**Pull request by mandan2**

### CODE REVIEW

File: src/Factory/ModuleFactory.php [hunk 1]
```diff
@@ -57,4 +57,15 @@ public function getModule(): ?Mollie
 
         return $module;
     }
+
+    public function getModuleName(): ?string
```
Comment (tomjas1997) [inline src/Factory/ModuleFactory.php:61]: Not sure if checking for null module really should happen. If module is null a lot more places will crash 🙃 overall would just get module via constructor and get name that way

Comment (tomjas1997) [reply src/Factory/ModuleFactory.php:61]: Also name would never be empty

File: src/ServiceProvider/BaseServiceProvider.php [hunk 2]
```diff
@@ -106,7 +106,8 @@ public function register(Container $container)
         $this->addService($container, MolCustomerRepository::class, MolCustomerRepository::class)
             ->withArgument('MolCustomer');
 
-        $this->addService($container, UninstallerInterface::class, Mollie\Install\DatabaseTableUninstaller::class);
+        $this->addService($container, UninstallerInterface::class, Mollie\Install\DatabaseTableUninstaller::class)
+            ->withArgument($container->get(ModuleFactory::class));
```
Comment (tomjas1997) [inline src/ServiceProvider/BaseServiceProvider.php:110]: Doesn't require any arguments here, it should work by auto-wiring

File: src/Factory/ModuleFactory.php (hunk 1 above)
Comment (mandan2) [reply src/Factory/ModuleFactory.php:61]: It doesn't hurt to keep this check as ide cries about return types if I don't check. :D  Anyway, this service won't be reached if module wouldn't exist.

File: src/ServiceProvider/BaseServiceProvider.php (hunk 2 above)
Comment (mandan2) [reply src/ServiceProvider/BaseServiceProvider.php]: Fixed

**Pull request merged**